module.go

```diff
@@ -391,3 +391,33 @@ func (m *Module) Iterator() *BPFObjectIterator {
 		prevMap:  nil,
 	}
 }
+
+func (m *Module) AttachProgs() error {
+	iters := m.Iterator()
+	for {
+		prog := iters.NextProgram()
+		if prog == nil {
+			break
+		}
+		link, err := prog.AttachGeneric()
+		if err != nil {
+			return err
+		}
+		m.links = append(m.links, link)
+	}
+
+	return nil
+}
+
+func (m *Module) DetachProgs() error {
```

Comment:
Shouldn't we remove the link from `m.links` when successfully detroyed?

Reply:
done

Reply:
How about not removing while iterating the slice? I know that the lib doesn't have multi-access control, but minimising possible races is better. As our Destroy returns error but DetachProgs should emulate the full destruction of bpf_object__detach_skeleton (that disregards errors), when a err occurs, we should store the non destroyed link with its error (maybe in a local map) and continue iterating.

At the end, assign an empty slice to m.links. If the local map is len > 0, iterate it reassigning the non destroyed links and wrapping/formatting all the errors to be returned.

WDYT?

Reply:
Thanks，your method seems better, i will update it latter.

Reply:
> Thanks，your method seems better, i will update it latter.

Done, check it out when you have time, thank you.